src/gui/processing/qgsprocessingmaplayercombobox.cpp

```diff
@@ -657,6 +702,31 @@ void QgsProcessingMapLayerComboBox::showSourceOptions()
   }
 }
 
+void QgsProcessingMapLayerComboBox::showRasterSourceOptions()
+{
+  if ( QgsPanelWidget *panel = QgsPanelWidget::findParentPanel( this ) )
+  {
+    QgsProcessingRasterSourceOptionsWidget *widget = new QgsProcessingRasterSourceOptionsWidget();
+    widget->setPanelTitle( tr( "%1 Options" ).arg( mParameter->description() ) );
+    widget->setReferenceScale( mRasterReferenceScale );
+    widget->setDpi( mRasterDpi );
```

Comment:
set the widget parameter capabilities here too, so that the widget can hide/show controls based on what the parameter capabilities are.